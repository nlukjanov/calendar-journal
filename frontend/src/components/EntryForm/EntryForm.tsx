import React from 'react';

import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';

type EntryForm = {
  formData: any;
  handleSubmit: any;
  handleChange: any;
};

const EntryForm = ({ formData, handleSubmit, handleChange }: EntryForm) => {
  console.log('formData', formData);
  return (
    <Card className='mx-auto mt-5 border-light' style={{ maxWidth: '25rem' }}>
      <Form onSubmit={handleSubmit}>
        <Form.Group controlId='title'>
          <Form.Label>Title</Form.Label>
          <Form.Control
            name='title'
            type='title'
            placeholder='Enter title'
            value={formData.title}
            onChange={handleChange}
          />
        </Form.Group>
        <Form.Group controlId='entryText'>
          <Form.Label>Entry Text</Form.Label>
          <Form.Control
            type='textarea'
            name='entryText'
            placeholder='Entry Text'
            value={formData.entryText}
            onChange={handleChange}
          />
        </Form.Group>
        <Form.Group controlId='date'>
          <Form.Label>Date</Form.Label>
          <Form.Control
            type='datetime-local'
            name='date'
            placeholder='Date'
            value={formData.date}
            onChange={handleChange}
          />
        </Form.Group>
        <Button variant='primary' type='submit'>
          Submit
        </Button>
      </Form>
    </Card>
  );
};

export default EntryForm;
