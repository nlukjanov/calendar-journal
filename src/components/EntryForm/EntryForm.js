import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import axios from 'axios';

const EntryForm = () => {
  const history = useHistory();
  const [formData, setFormData] = useState({ title: '', entryText: '' });

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      await axios.post('http://localhost:4000/api/journal', formData);
      history.push('/myjournal');
      // add notification with res
    } catch (error) {
      console.log(error);
      // setSignupErrors(error);
    }
  };

  const handleChange = ({ target: { name, value } }) => {
    const formNewData = { ...formData, [name]: value };
    setFormData(formNewData);
  };

  return (
    <form data-testid='entry-form' onSubmit={handleSubmit}>
      <label htmlFor='title'>Title</label>
      <input
        id='title'
        name='title'
        placeholder='Entry title'
        type='text'
        onChange={handleChange}
        value={formData.title}
      />
      <label htmlFor='entryText'>Entry text</label>
      <input
        id='entryText'
        name='entryText'
        placeholder='Entry text'
        type='text'
        onChange={handleChange}
        value={formData.entryText}
      />
      <button type='submit'>Create Entry</button>
    </form>
  );
};

export default EntryForm;
