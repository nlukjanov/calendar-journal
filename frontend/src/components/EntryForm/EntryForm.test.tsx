import React from 'react';

import { render } from '@testing-library/react';
import EntryForm from './EntryForm';

test('renders entry form correctly', async () => {
  const formData = {
    date: '',
    title: '',
    entryText: '',
  };
  const handleChange = jest.fn();
  const handleSubmit = jest.fn();
  const { container } = render(
    <EntryForm
      formData={formData}
      handleSubmit={handleSubmit}
      handleChange={handleChange}
    />,
  );
  expect(container).toMatchSnapshot();
});
