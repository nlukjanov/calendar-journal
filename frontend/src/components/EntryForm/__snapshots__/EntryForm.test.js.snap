// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders entry form correctly 1`] = `
<div>
  <div
    class="mx-auto mt-5 border-light card"
    style="max-width: 25rem;"
  >
    <form
      class=""
    >
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="title"
        >
          Title
        </label>
        <input
          class="form-control"
          id="title"
          name="title"
          placeholder="Enter title"
          type="title"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="entryText"
        >
          Entry Text
        </label>
        <input
          class="form-control"
          id="entryText"
          name="entryText"
          placeholder="Entry Text"
          type="textarea"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="date"
        >
          Date
        </label>
        <input
          class="form-control"
          id="date"
          name="date"
          placeholder="Date"
          type="datetime-local"
          value=""
        />
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
